### PR TITLE
[util] fixes #1235 - Response logged on HTTP error

### LINKED ISSUE #1235: Log http response error messages
Modify either the elapsed handler logger to add the response body when the status code is not 200, or inside the `Error400` etc methods, accept a logger argument and log the status and error message

### PR DESCRIPTION
**Pull request by olemis**
Fixes #1235 

Changes:
- Log HTTP error responses as errors while running integration tests .
- Attach response contents for HTTP errors in `http.response` log entry field .

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

**Comment by olemis** [comment]
screen shot 2018-04-18 at 9 53 27 pm


As can be seen in attached screenshot , contents of HTTP responses are binary . I am not sure this actually says anything meaningful at all . Is it better to convert it to a characters string ?

**Comment by gz-c** [comment]
@olemis yes as string, should probably use https://golang.org/pkg/bytes/#Buffer

**Comment by olemis** [comment]
@gz-c I'll both.

File: src/util/http/log.go [hunk 1]
```diff
@@ -16,7 +17,8 @@ func ElapsedHandler(logger logrus.FieldLogger, handler http.Handler) http.Handle
 		logMethod := logger.Infof
 		if lrw.statusCode >= 400 {
 			logMethod = logger.WithFields(logrus.Fields{
-				"http.response": lrw.response,
+				"http.response.bytes": lrw.response.Bytes(),
```
Comment (gz-c) [inline src/util/http/log.go]: dont log the bytes

File: src/util/http/log.go [hunk 2]
```diff
@@ -13,20 +14,32 @@ func ElapsedHandler(logger logrus.FieldLogger, handler http.Handler) http.Handle
...
+	return &wrappedResponseWriter{w, http.StatusOK, bytes.Buffer{}}
 }
 
 func (lrw *wrappedResponseWriter) WriteHeader(code int) {
 	lrw.statusCode = code
 	lrw.ResponseWriter.WriteHeader(code)
 }
+
+func (lrw *wrappedResponseWriter) Write(buff []byte) (int, error) {
+	lrw.response.Write(buff)
```
[the remark this answers is not in the document]
Comment (olemis) [reply src/util/http/log.go]: Good suggestion . Done !

**Pull request merged**